File: tools/parity/dispatch/agent-hook-endpoint-file.ts
// TS dispatch for the agent-hook-endpoint-file parity module: maps the shared
// vector function names to the real `src/shared/agent-hook-endpoint-file.ts`
// exports so the harness compares the live TS reference against the Rust port.

import { isAgentHookEndpointFileName } from '../../../src/shared/agent-hook-endpoint-file'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'isAgentHookEndpointFileName':
      return isAgentHookEndpointFileName(input as string)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-parity/src/modules/agent_hook_endpoint_file.rs
//! Parity dispatch for `orca_core::agent_hook_endpoint_file` vs
//! `src/shared/agent-hook-endpoint-file.ts`.

use orca_core::agent_hook_endpoint_file::is_agent_hook_endpoint_file_name;
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    match function {
        // `isAgentHookEndpointFileName(name: string)`: vectors only carry strings;
        // a non-string is treated as the empty name (never a known file name).
        "isAgentHookEndpointFileName" => {
            Value::Bool(is_agent_hook_endpoint_file_name(input.as_str().unwrap_or_default()))
        }
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}
